DOCS: DOCATT-8754: Updated TrackedDevice class documentation

unitALG opened the pull request:
### Description

Added documentation for the `InputSystem.TrackedDevice` class, particularly for the 4 properties. Several users had requested more information about these properties and their values, specifically the `IntegerControl` value in the `trackingState` property.

### Testing status & QA

No code was altered; documentation changes only. However, I did run the formatting tool and regenerated the documentation locally to test it.

### Overall Product Risks

Very low risk (documentation only).

### Comments to reviewers

I've requested a technical review from Chris Massie, who kindly provided me with some background on this class. 

### Checklist

- [x] Commit message for squash-merge is prefixed with `DOCS: ___`.

unitALG committed:
DOCATT-8754: Updated TrackedDevice class documentation

unitALG committed:
DOCATT-8754: Corrected position/pose in deviceRotation/devicePosition

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,85 @@ namespace UnityEngine.InputSystem
+    /// [Tracked Pose Driver](xref:input-system-tracked-input-devices#tracked-pose-driver)
+    /// component rather than being read directly from this class.
+    ///
+    /// Refer to the [Starter Assets](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@latest/index.html?subfolder=/manual/samples-spatial-keyboard.html)
```
```suggestion
    /// Refer to the [Starter Assets](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@latest/index.html?subfolder=/manual/samples-starter-assets.html)
```

amyloo22 replied:
I added the `com.unity.xr.interaction.toolkit` package to the projectMetadata.json file and used the xref id

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,85 @@ namespace UnityEngine.InputSystem
+        /// <remarks>
+        /// This property can contain a simple bit (1 or 0), but for some Open XR devices, this might
+        /// contain a float that includes bits that can indicate whether the device's position is actual
+        /// or inferred from its last-known [pose](xref:openxr-input#pose-data) (OpenXR Plugin package).
```
`xref:openxr-input#pose-data` doesn't seem to link correctly when I built the documentation. I think you'll have to edit the package's `projectMetadata.json` to add an `xref` tag for `"com.unity.xr.openxr"` for this to work.

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,85 @@ namespace UnityEngine.InputSystem
+        /// </summary>
+        /// <remarks>
+        /// This property can contain a simple bit (1 or 0), but for some Open XR devices, this might
+        /// contain a float that includes bits that can indicate whether the device's position is actual
```
I'm not sure we ever actually set this field to a float, I think it's always 0 or 1. The com.unity.xr.openxr package has existing documentation for [isTracked](https://github.cds.internal.unity3d.com/unity/xr.sdk.openxr/blob/4447698467d97ebc2e8348d31363a4e9b4667be2/com.unity.xr.openxr/Runtime/input/PoseControl.cs#L19) and it doesn't indicate that we treat it as anything other than a bool.

amyloo22 replied:
Thanks, I'll update this. It is a little confusing because the `TrackedDevice.isTracked` property returns a `ButtonControl` not a `bool`.

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,85 @@ namespace UnityEngine.InputSystem
+        /// or inferred from its last-known [pose](xref:openxr-input#pose-data) (OpenXR Plugin package).
+        ///
+        /// For more information about how the float represents inferred position vs. actual position in
+        /// OpenXR devices, refer to [Reference Spaces](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#reference-spaces)
```
It might be better to use the newer 1.1 version of the OpenXR spec: https://registry.khronos.org/OpenXR/specs/1.1/html/xrspec.html#spaces-reference-spaces

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,85 @@ namespace UnityEngine.InputSystem
+        /// Indicates whether the input device is actively tracked or not.
+        /// </summary>
+        /// <remarks>
+        /// This property can contain a simple bit (1 or 0), but for some Open XR devices, this might
```
```suggestion
        /// This property can contain a simple bit (1 or 0), but for some OpenXR devices, this might
```
OpenXR doesn't have a space.

unitALG committed:
DOCATT-8754: Post tech-review updates

- Fixed some links, clarified isTracked return in TrackedDevice.cs
- Added related XR packages to `xref` list in projectMetadata.json

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,81 @@ namespace UnityEngine.InputSystem
+
+        /// <summary>
+        /// Represents the position portion of the input device's primary
+        /// [pose](xref:input-system-tracked-input-devices#tracked-pose-driver). For an HMD
```
```suggestion
        /// [pose](xref:openxr-input#pose-data). For an HMD
```

chris-massie commented on Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs:
```diff
@@ -7,17 +7,81 @@ namespace UnityEngine.InputSystem
+
+        /// <summary>
+        /// Represents the rotation portion of the input device's primary
+        /// [pose](xref:input-system-tracked-input-devices#tracked-pose-driver). For an HMD
```
```suggestion
        /// [pose](xref:openxr-input#pose-data). For an HMD
```

chris-massie reviewed (approved):
I gave a suggestion to replace a Pose link to go to https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.16/manual/input.html#pose-data instead of the Tracked Pose Driver component documentation since it's more direct to what the pose data is.

unitALG committed:
Update Packages/com.unity.inputsystem/InputSystem/Devices/TrackedDevice.cs

Co-authored-by: Chris Massie <67029035+chris-massie@users.noreply.github.com>
